common/src/main/java/org/keycloak/common/util/CollectionUtil.java

```diff
@@ -75,4 +78,15 @@ public static boolean isEmpty(Collection<?> collection) {
     public static boolean isNotEmpty(Collection<?> collection) {
         return !isEmpty(collection);
     }
+
+    public static <T> Set<T> intersection(Collection<T> col1, Collection<T> col2) {
+        if (isEmpty(col1) || isEmpty(col2)) return Collections.emptySet();
+
+        final Collection<T> iteratorCollection = col1.size() <= col2.size() ? col1 : col2;
+        final Collection<T> searchCollection = iteratorCollection.equals(col1) ? col2 : col1;
+
+        return iteratorCollection.stream()
+                .filter(searchCollection::contains)
+                .collect(Collectors.toSet());
+    }
```

Comment:
I think we could make this generic? I'm not 100% sure this works though. 😄
```suggestion
    public static <T> Collection<T> intersection(Collection<T> col1, Collection<T> col2) {
        if (isEmpty(col1) || isEmpty(col2)) return Collections.emptySet();

        final Collection<T> iteratorCollection = col1.size() <= col2.size() ? col1 : col2;
        final Collection<T> searchCollection = iteratorCollection.equals(col1) ? col2 : col1;

        return iteratorCollection.stream()
                .filter(searchCollection::contains)
                .collect(Collectors.toCollection());
    }
```

Reply:
@vmuzikar I'd keep it as is. I'd say the intersection should contain only unique items.